--- ./boost/safe_numerics/exception.hpp
+++ ./boost/safe_numerics/exception.hpp
@@ -45,7 +45,7 @@
     uninitialized_value         // creating of uninitialized value
 };
 
-const char * literal_string(const safe_numerics_error & e){
+inline const char * literal_string(const safe_numerics_error & e){
     switch(e){
     case safe_numerics_error::success: return "success";
     case safe_numerics_error::positive_overflow_error: return "positive_overflow_error";
